internal/services/admin/v1/server.go

```diff
@@ -0,0 +1,176 @@
+package v1
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	msgqueue "github.com/hatchet-dev/hatchet/internal/msgqueue/v1"
+	contracts "github.com/hatchet-dev/hatchet/internal/services/admin/contracts/v1"
+	tasktypes "github.com/hatchet-dev/hatchet/internal/services/shared/tasktypes/v1"
+	"github.com/hatchet-dev/hatchet/pkg/repository/postgres/dbsqlc"
+	"github.com/hatchet-dev/hatchet/pkg/repository/postgres/sqlchelpers"
+	v1 "github.com/hatchet-dev/hatchet/pkg/repository/v1"
+	"github.com/hatchet-dev/hatchet/pkg/repository/v1/sqlcv1"
+)
+
+func (a *AdminServiceImpl) CancelTasks(ctx context.Context, req *contracts.CancelTasksRequest) (*contracts.CancelTasksResponse, error) {
+	tenant := ctx.Value("tenant").(*dbsqlc.Tenant)
+
+	tasksToCancel := []v1.TaskIdRetryCount{}
+
+	if len(req.Tasks) == 0 {
+		return &contracts.CancelTasksResponse{}, nil
+	}
+
+	// for tasks passed in directly, make sure the tenant id and retry counts match
+	if len(req.Tasks) > 0 {
```

Comment:
nit: do we need this `if` here? I'd think if we get this point, then we know this condition is `true`

Reply:
ah, I actually meant to remove the `if len(req.Tasks) == 0` check above, because a user can pass both a filter and a list of tasks with ids.